### PR TITLE
Update dependencies 

### PR DESCRIPTION
**Pull request by hanzei**
- Bump Go to 1.14
- Bump golangci-lint version to 1.24.0

### CODE REVIEW

File: plugin-ci/orb.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ orbs:
 executors:
   default:
     docker:
-      - image: circleci/golang:1.13-node
+      - image: circleci/golang:1.14-node
```
Comment (agnivade) [inline plugin-ci/orb.yml]: Let's use 1.14.1 to pin down to a specific version.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: plugin-ci/orb.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ commands:
     parameters:
       version:
         type: string
-        default: 1.21.0
+        default: 1.24.0
```
Comment (hanzei) [inline plugin-ci/orb.yml:33]: @agnivade For my understanding: There is no way for a plugin that uses this orb to specify the golangci-lint version number, right?

Comment (agnivade) [reply plugin-ci/orb.yml:33]: They can actually. This is just the default value if they choose not to override. See the .circleci/config.yml file in the mmctl repo.

**Review by agnivade — APPROVED** [review]

File: plugin-ci/orb.yml (hunk 2 above)
Comment (hanzei) [reply plugin-ci/orb.yml:33]: I see that a consumer of the orb can set the version for `golangci-lint` command, but how can this be one for a consumer of the `lint` job? Could you provide an example?

Comment (agnivade) [reply plugin-ci/orb.yml:33]: Oh you know what, scratch what I said. mmctl just parameterizes the installation. Yeah, I think you are right. Although I would have to look up circleCI documentation to be totally sure.

**Pull request merged**